File: bmlsj/01/귤고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        HashMap<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < tangerine.length; i++) {
            map.put(tangerine[i], map.getOrDefault(tangerine[i], 0) + 1);
        }

        List<Integer> keySet = new ArrayList<>(map.keySet());
        keySet.sort(new Comparator<Integer>() {
            @Override
            public int compare(Integer o1, Integer o2) {
                return map.get(o2) - map.get(o1);
            }
        });

        int total = 0;
        for (int key : keySet) {
            total += map.get(key);
            answer++;
            if (total >= k) {
                break;
            }
        }

        return answer;
    }
}

File: bmlsj/01/귤고르기.py
from collections import Counter

def solution(k, tangerine):
    answer = 0
    
    counter = Counter(tangerine)
    counts = sorted(counter.values(), reverse=True)
    
    total = 0
    for c in counts:
        total += c
        answer += 1
        if total >= k:
            break
    
    return answer